preprocessor/src/main/java/com/azure/autorest/preprocessor/tranformer/Transformer.java

```diff
@@ -310,86 +313,97 @@ private void addPagingNextOperation(Client client, OperationGroup operationGroup
+        requestParams.add(nextLink);
+        nextOperation.getRequests().get(0).setParameters(requestParams);
+        List<Parameter> signatureParams = new ArrayList<>();
+        signatureParams.add(nextLink);
+        nextOperation.getRequests().get(0).setSignatureParameters(signatureParams);
+        nextOperation.setApiVersions(operation.getApiVersions());
+        nextOperation.setDeprecated(operation.getDeprecated());
+        nextOperation.setDescription(operation.getDescription());
+        nextOperation.setExceptions(operation.getExceptions());
+        nextOperation.setExtensions(operation.getExtensions());
+        nextOperation.setExternalDocs(operation.getExternalDocs());
+        nextOperation.setProfile(operation.getProfile());
+        nextOperation.setResponses(operation.getResponses());
+        nextOperation.setSummary(operation.getSummary());
+        nextOperation.setUid(operation.getUid());
+
+        Operation nextOperationLocal = nextOperation;
+
+        if (operation.getExtensions().getXmsPageable().getOperationName() == null) {
+          operation.getRequests().stream().flatMap(r -> r.getParameters().stream())
+                  .filter(parameter -> {
+                    return parameter.getProtocol() == null || parameter.getProtocol().getHttp() == null
+                            || (parameter.getProtocol().getHttp().getIn() != null
+                            && (parameter.getProtocol().getHttp().getIn().equals(RequestParameterLocation.HEADER)
+                            || parameter.getProtocol().getHttp().getIn().equals(RequestParameterLocation.URI)));
+                  })
+                  .forEach(param -> {
+                    nextOperationLocal.getRequests().get(0).getParameters().add(param);
+                  });
+
+          operation.getRequests().stream().flatMap(r -> r.getSignatureParameters().stream())
+                  .filter(parameter -> {
+                    return parameter.getProtocol() == null || parameter.getProtocol().getHttp() == null
+                            || (parameter.getProtocol().getHttp().getIn() != null
+                            && (parameter.getProtocol().getHttp().getIn().equals(RequestParameterLocation.HEADER)
+                            || parameter.getProtocol().getHttp().getIn().equals(RequestParameterLocation.URI)));
+                  })
+                  .forEach(param -> {
+                    nextOperationLocal.getRequests().get(0).getSignatureParameters().add(param);
+                  });
+        }
+        operation.getExtensions().getXmsPageable().setNextOperation(nextOperation);
+        nextOperation.getExtensions().getXmsPageable().setNextOperation(nextOperation);
+        operationNextOperationMap.put(operationSignature, nextOperation);
+      } else {
+        // Same pageable operation should always share the same next link operation
+        nextOperation = operationNextOperationMap.get(operationSignature);
```

Comment:
Change here. 

Current implementation shares `operation.extension` across all generated NextOperation for the same operation. When the operation is processed multiple times(in different operation groups), previous generated NextOperation's `nextOperation` will be set to the newly generated one, causing it to be considered normal pageable operation. Hence `listNext` is generated in service client.